ADBDEV-4904-26: Remove the isnull check for NULL.

red1452 opened the pull request:
Remove the isnull check for NULL.

The slot_getsysattr function is always called with the isnull argument not equal
to NULL, so I changed the isnull comparison with NULL to an assertion and also
added an assertion to the heap_getsysattr function.

red1452 committed:
Add check isnull is not NULL to the heap_getsysattr function

The isnull pointer may not be NULL at the heap_getsysattr, but at the same
function slot_getsysattr there is check "isnull is not NULL". It says, that this
pointer may be NULL, but now the slot_getsysattr function and the
heap_getsysattr function are called with the isnull attribute, which is not
NULL.

RekGRpth commented:
I think you should replace the `isnull` check with NULL in the `slot_getsysattr` function with an assertion because it is never called with a null last argument.

RekGRpth commented:
add title to PR description

RekGRpth commented:
> At the slot_getsysattr there is check (`isnull` id not NULL) before using.

So I propose replacing this condition with an assertion.

red1452 commented:
> add title to PR description

done.

red1452 committed:
change check to assert

RekGRpth commented:
> > At the slot_getsysattr there is check (`isnull` id not NULL) before using.
> 
> So I propose replacing this condition with an assertion.

I suggest replace https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/execTuples.c#L1547-L1548 with
```c
        Asssert(isnull != NULL);
        *isnull = false;
```

RekGRpth committed:
Remove the isnull check for NULL.

The slot_getsysattr function is always called with the isnull argument not equal
to NULL, so I changed the isnull comparison with NULL to an assertion and also
added an assertion to the heap_getsysattr function.

bandetto commented on src/backend/access/common/heaptuple.c:
```diff
@@ -556,6 +556,7 @@ heap_getsysattr(HeapTuple tup, int attnum, TupleDesc tupleDesc, bool *isnull)
 
 	Assert(tup);
 	Assert(!is_memtuple((GenericTuple) tup));
+	Assert(isnull);
```
```suggestion
	Assert(isnull != NULL);
```
To be consistent

RekGRpth replied:
applied

RekGRpth committed:
Update src/backend/access/common/heaptuple.c

Co-authored-by: Vladimir Sarmin <v.sarmin@arenadata.io>
